From 5f1ac685992183d71097669b1956b2af06ad6ed0 Mon Sep 17 00:00:00 2001
From: Nathan Owens <ndowens@artixlinux.org>
Date: Tue, 29 Dec 2020 19:44:34 -0600
Subject: [PATCH] fix cross build

Replace AC_CHECK_FILES as it is not cross-friendly
---
diff --git configure.ac configure.ac
index d8a7cf1..31c1245 100644
--- configure.ac
+++ configure.ac
@@ -214,7 +214,7 @@ if test x$enable_sse = xyes; then
       AX_CHECK_COMPILER_FLAGS(-msse, [SBSMS_CFLAGS="$SBSMS_CFLAGS -msse"],[AC_MSG_ERROR([Need a version of gcc with -msse])])
    fi
 
-   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <xmmintrin.h>]], [[__m128 v,w; float p[8]; v = _mm_loadu_ps(p); _mm_storeu_ps(p,v); w = _mm_loadl_pi(w,(const __m64*)p); w = _mm_loadh_pi(w,(const __m64*)p); _mm_storel_pi((__m64 *)p, w); _mm_storeh_pi((__m64 *)p, w); v = _mm_add_ps(v,w); v = _mm_sub_ps(v,w); v = _mm_mul_ps(v,w); v = _mm_shuffle_ps(v,w,_MM_SHUFFLE(0,1,2,3)); w = _mm_set1_ps(0.0f);]])], [sse_ok=yes], [sse_ok=no])
+   AC_TRY_COMPILE([AC_LANG_PROGRAM([[#include <xmmintrin.h>]], [[__m128 v,w; float p[8]; v = _mm_loadu_ps(p); _mm_storeu_ps(p,v); w = _mm_loadl_pi(w,(const __m64*)p); w = _mm_loadh_pi(w,(const __m64*)p); _mm_storel_pi((__m64 *)p, w); _mm_storeh_pi((__m64 *)p, w); v = _mm_add_ps(v,w); v = _mm_sub_ps(v,w); v = _mm_mul_ps(v,w); v = _mm_shuffle_ps(v,w,_MM_SHUFFLE(0,1,2,3)); w = _mm_set1_ps(0.0f);]])], [sse_ok=yes], [sse_ok=no])
 
    if test x$sse_ok = xyes; then
       AC_DEFINE(ENABLE_SSE,1,[Define to enable sse])
diff --git m4/audacity_checklib_ffmpeg.m4 m4/audacity_checklib_ffmpeg.m4
index b98110f..d4754aa 100644
--- m4/audacity_checklib_ffmpeg.m4
+++ m4/audacity_checklib_ffmpeg.m4
@@ -35,13 +35,9 @@ AC_DEFUN([AUDACITY_CHECKLIB_FFMPEG], [
 
    dnl see if ffmpeg is available locally, or rather that we have some headers
    dnl in lib-src/ffmpeg/ we can use.
-   AC_CHECK_FILE(${srcdir}/lib-src/ffmpeg/libavcodec/avcodec.h,
-                 avcodec_h_found="yes",
-                 avcodec_h_found="no")
+   AS_IF([test -f ${srcdir}/lib-src/ffmpeg/libavcodec/avcodec.h], [avcodec_h_found=yes])
 
-   AC_CHECK_FILE(${srcdir}/lib-src/ffmpeg/libavformat/avformat.h,
-                 avformat_h_found="yes",
-                 avformat_h_found="no")
+   AS_IF([test -f ${srcdir}/lib-src/ffmpeg/libavformat/avformat.h], [avformat_h_found=yes])
 
    if test "$avcodec_h_found" = "yes" -a "$avformat_h_found" = "yes"; then
       FFMPEG_LOCAL_AVAILABLE="yes"
diff --git m4/audacity_checklib_lame.m4 m4/audacity_checklib_lame.m4
index 28ecd77..5794143 100644
--- m4/audacity_checklib_lame.m4
+++ m4/audacity_checklib_lame.m4
@@ -33,9 +33,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LAME], [
 
    dnl see if LAME is available in the source dir
 
-   AC_CHECK_FILE(${srcdir}/lib-src/lame/lame/lame.h,
-                 LAME_LOCAL_AVAILABLE="yes",
-                 LAME_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/lame/lame/lame.h], [LAME_LOCAL_AVAILABLE=yes])
 
    if test "$LAME_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([LAME headers are available in this source tree.])
diff --git m4/audacity_checklib_libexpat.m4 m4/audacity_checklib_libexpat.m4
index c0c88db..ffe76ac 100644
--- m4/audacity_checklib_libexpat.m4
+++ m4/audacity_checklib_libexpat.m4
@@ -42,9 +42,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_EXPAT], [
 
    dnl see if expat is available in the local tree
 
-   AC_CHECK_FILE(${srcdir}/lib-src/expat/lib/expat.h,
-                 EXPAT_LOCAL_AVAILABLE="yes",
-                 EXPAT_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/expat/lib/expat.h], [EXPAT_LOCAL_AVAILABLE=yes])
 
    if test "$EXPAT_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([Expat libraries are available in the local tree])
diff --git m4/audacity_checklib_libflac.m4 m4/audacity_checklib_libflac.m4
index 1002986..6096791 100644
--- m4/audacity_checklib_libflac.m4
+++ m4/audacity_checklib_libflac.m4
@@ -38,13 +38,9 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBFLAC], [
 
    dnl see if FLAC is available in the source dir
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libflac/include/FLAC/format.h,
-                 flac_h_available="yes",
-                 flac_h_available="no")
+   AS_IF([test -f ${srcdir}/lib-src/libflac/include/FLAC/format.h], [flac_h_available=yes])
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libflac/include/FLAC++/decoder.h,
-                 flacpp_h_available="yes",
-                 flacpp_h_available="no")
+   AS_IF([test -f ${srcdir}/lib-src/libflac/include/FLAC++/decoder.h], [flacpp_h_available=yes])
 
    if test "$flac_h_available" = "yes" -a "$flacpp_h_available" = "yes"; then
       LIBFLAC_LOCAL_AVAILABLE="yes"
diff --git m4/audacity_checklib_libid3tag.m4 m4/audacity_checklib_libid3tag.m4
index b69c2cb..59e94cc 100644
--- m4/audacity_checklib_libid3tag.m4
+++ m4/audacity_checklib_libid3tag.m4
@@ -25,9 +25,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBID3TAG], [
 
    dnl see if libid3tag is available in the local tree
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libid3tag/frame.h,
-                 LIBID3TAG_LOCAL_AVAILABLE="yes",
-                 LIBID3TAG_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/libid3tag/frame.h], [LIBID3TAG_LOCAL_AVAILABLE=yes])
 
    if test "$LIBID3TAG_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([libid3tag libraries are available in the local tree])
diff --git m4/audacity_checklib_libmad.m4 m4/audacity_checklib_libmad.m4
index 0c2d02a..d53c459 100644
--- m4/audacity_checklib_libmad.m4
+++ m4/audacity_checklib_libmad.m4
@@ -34,9 +34,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBMAD], [
 
    dnl see if libmad is available in the local tree
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libmad/frame.h,
-                 LIBMAD_LOCAL_AVAILABLE="yes",
-                 LIBMAD_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/libmad/frame.h], [LIBMAD_LOCAL_AVAILABLE=yes])
 
    if test "$LIBMAD_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([libmad libraries are available in the local tree])
diff --git m4/audacity_checklib_libnyquist.m4 m4/audacity_checklib_libnyquist.m4
index 3f1c98f..b4af690 100644
--- m4/audacity_checklib_libnyquist.m4
+++ m4/audacity_checklib_libnyquist.m4
@@ -17,9 +17,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBNYQUIST], [
 
    dnl see if Nyquist is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libnyquist/nyx.h,
-                 LIBNYQUIST_LOCAL_AVAILABLE="yes",
-                 LIBNYQUIST_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/libnyquist/nyx.h], [LIBNYQUIST_LOCAL_AVAILABLE=yes])
 
    if test "$LIBNYQUIST_LOCAL_AVAILABLE" = "yes" ; then
       AC_MSG_NOTICE([nyquist libraries are available in the local tree])
diff --git m4/audacity_checklib_libsbsms.m4 m4/audacity_checklib_libsbsms.m4
index c2f80dd..a0ec94b 100644
--- m4/audacity_checklib_libsbsms.m4
+++ m4/audacity_checklib_libsbsms.m4
@@ -25,9 +25,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBSBSMS], [
 
    dnl see if libsbsms is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/sbsms/include/sbsms.h,
-                 LIBSBSMS_LOCAL_AVAILABLE="yes",
-                 LIBSBSMS_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/sbsms/include/sbsms.h], [LIBSBSMS_LOCAL_AVAILABLE=yes])
 
    if test "$LIBSBSMS_LOCAL_AVAILABLE" = "yes"; then
       dnl do not build programs we don't need
diff --git m4/audacity_checklib_libsndfile.m4 m4/audacity_checklib_libsndfile.m4
index d49e149..e501899 100644
--- m4/audacity_checklib_libsndfile.m4
+++ m4/audacity_checklib_libsndfile.m4
@@ -25,9 +25,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBSNDFILE], [
 
    dnl see if libsndfile is available in the local tree
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libsndfile/src/sndfile.h.in,
-                 LIBSNDFILE_LOCAL_AVAILABLE="yes",
-                 LIBSNDFILE_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/libsndfile/src/sndfile.h.in], [LIBSNDFILE_LOCAL_AVAILABLE=yes])
 
    if test "$LIBSNDFILE_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([libsndfile libraries are available in this source tree])
diff --git m4/audacity_checklib_libsoundtouch.m4 m4/audacity_checklib_libsoundtouch.m4
index 0b8aa69..d8d9f58 100644
--- m4/audacity_checklib_libsoundtouch.m4
+++ m4/audacity_checklib_libsoundtouch.m4
@@ -40,9 +40,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBSOUNDTOUCH], [
 
    dnl see if libsoundtouch is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/soundtouch/include/SoundTouch.h,
-                 LIBSOUNDTOUCH_LOCAL_AVAILABLE="yes",
-                 LIBSOUNDTOUCH_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/soundtouch/include/SoundTouch.h], [LIBSOUNDTOUCH_LOCAL_AVAILABLE=yes])
 
    if test "$LIBSOUNDTOUCH_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([libsoundtouch libraries are available in the local tree])
diff --git m4/audacity_checklib_libsoxr.m4 m4/audacity_checklib_libsoxr.m4
index a331e9b..1a0702b 100644
--- m4/audacity_checklib_libsoxr.m4
+++ m4/audacity_checklib_libsoxr.m4
@@ -25,9 +25,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBSOXR], [
 
    dnl see if libsoxr is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libsoxr/src/soxr.h,
-                 LIBSOXR_LOCAL_AVAILABLE="yes",
-                 LIBSOXR_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/libsoxr/src/soxr.h], [LIBSOXR_LOCAL_AVAILABLE=yes])
 
    if test "$LIBSOXR_LOCAL_AVAILABLE" = "yes"; then
       # Breaks other other libraries in Audacity tree; but why is ./configure
diff --git m4/audacity_checklib_libtwolame.m4 m4/audacity_checklib_libtwolame.m4
index 2d28b07..fc2a2bf 100644
--- m4/audacity_checklib_libtwolame.m4
+++ m4/audacity_checklib_libtwolame.m4
@@ -25,9 +25,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBTWOLAME], [
 
    dnl see if libtwolame is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/twolame/libtwolame/twolame.h,
-                 LIBTWOLAME_LOCAL_AVAILABLE="yes",
-                 LIBTWOLAME_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/twolame/libtwolame/twolame.h], [LIBTWOLAME_LOCAL_AVAILABLE=yes])
 
    if test "$LIBTWOLAME_LOCAL_AVAILABLE" = "yes"; then
       dnl disable programs we don't need to build
diff --git m4/audacity_checklib_libvamp.m4 m4/audacity_checklib_libvamp.m4
index 8a1e0e1..57dc5eb 100644
--- m4/audacity_checklib_libvamp.m4
+++ m4/audacity_checklib_libvamp.m4
@@ -25,9 +25,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBVAMP], [
 
    dnl see if Vamp is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libvamp/vamp-hostsdk/PluginLoader.h,
-                 LIBVAMP_LOCAL_AVAILABLE="yes",
-                 LIBVAMP_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/libvamp/vamp-hostsdk/PluginLoader.h], [LIBVAMP_LOCAL_AVAILABLE=yes])
 
    if test "$LIBVAMP_LOCAL_AVAILABLE" = "yes"; then
       LIBVAMP_LOCAL_CONFIGURE_ARGS="--disable-programs"
diff --git m4/audacity_checklib_libvorbis.m4 m4/audacity_checklib_libvorbis.m4
index 0839fa9..3d8b70d 100644
--- m4/audacity_checklib_libvorbis.m4
+++ m4/audacity_checklib_libvorbis.m4
@@ -30,13 +30,9 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBVORBIS], [
 
    dnl see if Vorbis is available in the source dir
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libvorbis/include/vorbis/vorbisenc.h,
-                 vorbisenc_h_available="yes",
-                 vorbisenc_h_available="no")
+   AS_IF([test -f ${srcdir}/lib-src/libvorbis/include/vorbis/vorbisenc.h], [vorbisenc_h_available=yes])
 
-   AC_CHECK_FILE(${srcdir}/lib-src/libogg/include/ogg/ogg.h,
-                 ogg_h_available="yes",
-                 ogg_h_available="no")
+   AS_IF([test -f ${srcdir}/lib-src/libogg/include/ogg/ogg.h], [ogg_h_available=yes])
 
    if test "$vorbisenc_h_available" = "yes" -a "$ogg_h_available" = "yes"; then
       LIBVORBIS_LOCAL_AVAILABLE="yes"
diff --git m4/audacity_checklib_lv2.m4 m4/audacity_checklib_lv2.m4
index 78a279a..164aa6d 100644
--- m4/audacity_checklib_lv2.m4
+++ m4/audacity_checklib_lv2.m4
@@ -23,9 +23,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LV2], [
 
    dnl see if LV2 is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/lv2/configure,
-                 LV2_LOCAL_AVAILABLE="yes",
-                 LV2_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/lv2/configure], [LV2_LOCAL_AVAILABLE=yes])
 
    if test "$LV2_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([LV2 libraries are available in the local tree])
diff --git m4/audacity_checklib_portaudio.m4 m4/audacity_checklib_portaudio.m4
index 9b7674e..03d62b8 100644
--- m4/audacity_checklib_portaudio.m4
+++ m4/audacity_checklib_portaudio.m4
@@ -31,9 +31,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_PORTAUDIO], [
 
    dnl see if portaudio is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/portaudio-v19/include/portaudio.h,
-                 PORTAUDIO_LOCAL_AVAILABLE="yes",
-                 PORTAUDIO_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/portaudio-v19/include/portaudio.h], [PORTAUDIO_LOCAL_AVAILABLE=yes])
 
    if test "$PORTAUDIO_LOCAL_AVAILABLE" = "yes"; then
       dnl We need to override the pkg-config check for portmixer by passing
diff --git m4/audacity_checklib_portmidi.m4 m4/audacity_checklib_portmidi.m4
index 4e63b1b..6db1dc6 100644
--- m4/audacity_checklib_portmidi.m4
+++ m4/audacity_checklib_portmidi.m4
@@ -21,9 +21,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_PORTMIDI], [
       AC_MSG_NOTICE([portmidi library is NOT available as system library])
    fi
 
-   AC_CHECK_FILE(${srcdir}/lib-src/portmidi/pm_common/portmidi.h,
-                 PORTMIDI_LOCAL_AVAILABLE="yes",
-                 PORTMIDI_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/portmidi/pm_common/portmidi.h], [PORTMIDI_LOCAL_AVAILABLE=yes])
 
    if test "$PORTMIDI_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([portmidi library is available in the local tree])
diff --git m4/audacity_checklib_portsmf.m4 m4/audacity_checklib_portsmf.m4
index 9c336d8..3a31ef5 100644
--- m4/audacity_checklib_portsmf.m4
+++ m4/audacity_checklib_portsmf.m4
@@ -24,9 +24,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_PORTSMF], [
       AC_MSG_NOTICE([portSMF library is NOT available as system library])
    fi
 
-   AC_CHECK_FILE(${srcdir}/lib-src/portsmf/allegro.h,
-                 PORTSMF_LOCAL_AVAILABLE="yes",
-                 PORTSMF_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/portsmf/allegro.h], [PORTSMF_LOCAL_AVAILABLE=yes])
 
    if test "$PORTSMF_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([portSMF library is available in the local tree])
diff --git m4/audacity_checklib_widgetextra.m4 m4/audacity_checklib_widgetextra.m4
index 4ef0f97..f857723 100644
--- m4/audacity_checklib_widgetextra.m4
+++ m4/audacity_checklib_widgetextra.m4
@@ -28,9 +28,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_WIDGETEXTRA], [
 
    dnl see if libwidgetextra is available locally
 
-   AC_CHECK_FILE(${srcdir}/lib-src/lib-widget-extra/NonGuiThread.h,
-                 WIDGETEXTRA_LOCAL_AVAILABLE="yes",
-                 WIDGETEXTRA_LOCAL_AVAILABLE="no")
+   AS_IF([test -f ${srcdir}/lib-src/lib-widget-extra/NonGuiThread.h], [WIDGETEXTRA_LOCAL_AVAILABLE=yes])
 
    if test "$WIDGETEXTRA_LOCAL_AVAILABLE" = "yes"; then
       AC_MSG_NOTICE([libwidgetextra library is available in the local tree])
-- 
2.30.0

diff --git configure.ac configure.ac
index 31c1245..166a77e 100644
--- configure.ac
+++ configure.ac
@@ -63,7 +63,7 @@ AC_PROG_CXXCPP
 AM_PROG_AR
 
 LT_INIT([disable-shared])
-
+m4_pattern_allow([^AC_LANG])
 AC_PROG_INSTALL
 dnl pkg-config is required for lots of things
 AC_CHECK_PROG(HAVE_PKG_CONFIG, pkg-config, yes, no)
